account_avatax_demo_oca/tests/test_sale_order.py

```diff
@@ -0,0 +1,138 @@
+                "tax_code_id": cls.env.ref(
+                    "account_avatax_oca.avatax_product_taxcodeNT"
+                ).id,
+            }
+        )
+
+        cls.taxed_product = cls.env["product.product"].create(
+            {
+                "name": "Taxed Product",
+                "list_price": 33,
+                "sale_ok": True,
+            }
+        )
+
+        cls.partner = cls.env["res.partner"].create(
+            {
+                "name": "Test Partner",
+                "is_company": True,
+                "street": "77 Santa Barbara Rd",
+                "city": "Pleasant Hill",
+                "state_id": cls.env.ref("base.state_us_5").id,
+                "country_id": cls.env.ref("base.us").id,
+                "zip": "94523",
+            }
+        )
+
+        # Create sale order
+        cls.order = cls.env["sale.order"].create(
+            {
+                "company_id": cls.company.id,
+                "partner_id": cls.partner.id,
+            }
+        )
+        cls.uom_unit = cls.env.ref("uom.product_uom_unit")
+        cls.order.write(
+            {
+                "order_line": [
+                    (
+                        0,
+                        False,
+                        {
+                            "product_id": cls.NT_product.id,
+                            "name": "NT Product",
+                            "product_uom": cls.uom_unit.id,
+                            "product_uom_qty": 1.0,
+                        },
+                    ),
+                    (
+                        0,
+                        False,
+                        {
+                            "product_id": cls.taxed_product.id,
+                            "name": "Taxed Product",
+                            "product_uom": cls.uom_unit.id,
+                            "product_uom_qty": 1.0,
+                        },
+                    ),
+                ]
+            }
+        )
+
+        return res
+
+    def read_json(self, file_name):
+        module_path = os.path.dirname(__file__)
+        file_path = os.path.join(module_path, file_name)
+        with open(file_path) as file:
+            data = json.load(file)
+            return data
+
+    def test_compute_taxes_for_quotation(self):
+        """
+        Order:
+            -   Order Line 1: NT Product
+            -   Order Line 2: Taxed Product
+            -   Exemption Code: None
+        Expect:
+            -   Only order line 2 has tax applied
+        """
+        so_2_response = self.read_json("SaleOrder_response.json")
+        for line in so_2_response["lines"]:
+            if line["description"] == "NT Product":
+                line["lineNumber"] = self.order.order_line[0].id
+
+            if line["description"] == "Taxed Product":
+                line["lineNumber"] = self.order.order_line[1].id
+
+        with self._capture_create_or_adjust_transaction(return_value=so_2_response):
```

Comment:
```suggestion
        with self._capture_create_or_adjust_transaction(return_value=so_2_response) as captured:
```
capture the mock response to check the return values

Reply:
test both cases: 
- Case 1: with self._capture_create_or_adjust_transaction(return_value=so_2_response) as captured
- Case 2: with self._capture_create_or_adjust_transaction(return_value=so_2_response, apply_args=True) as captured

Each case, test the values of:
- captured.mock_response.json()["lines"][...]["lineNumber"]
- captured.mock_response.json()["lines"][...]["itemCode"]
- captured.mock_response.json()["type"]
- captured.mock_response.json()["customerCode"]